curriculum/challenges/english/25-front-end-development/quiz-javascript-loops/66edcd49e73385dd4df54ac7.md

```diff
@@ -17,439 +17,517 @@ Answer all of the questions below correctly to pass the quiz.
 
 ---
 
-Placeholder distractor 2
+`do...while` loop
 
 ---
 
-Placeholder distractor 3
+`for` loop
 
 #### --answer--
 
-Placeholder answer
+`for...in` loop
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the difference between the `for...in` loop and the `for...of` loop?
 
 #### --distractors--
 
-Placeholder distractor 1
+`for...in` loop iterates over property values, while the `for...of` loop iterates over property names (keys).
 
 ---
 
-Placeholder distractor 2
+`for...in` loops create new properties, while `for...of` loops modify existing properties.
 
 ---
 
-Placeholder distractor 3
+`for...in` loops can only be used with strings, while the `for...of` loops can be used with both strings and numbers.
 
 #### --answer--
 
-Placeholder answer
+`for...in` loop iterates over property names (keys), while the `for...of` loop iterates over property values.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+To prevent an infinite loop from occurring, all but one of these are expected to be part of the loop used.
 
 #### --distractors--
 
-Placeholder distractor 1
+Initialization of variable.
 
 ---
 
-Placeholder distractor 2
+A loop condition that eventually gives false.
 
 ---
 
-Placeholder distractor 3
+Increment/decrement logic.
 
 #### --answer--
 
-Placeholder answer
+A loop condition that always returns true.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What statement can be used to break out of a loop completely?
 
 #### --distractors--
 
-Placeholder distractor 1
+The `continue` statement
 
 ---
 
-Placeholder distractor 2
+The `label` statement
 
 ---
 
-Placeholder distractor 3
+The `skip` statement
 
 #### --answer--
 
-Placeholder answer
+The `break` statement
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which statement can be used to skip the current iteration and move to the next iteration?
 
 #### --distractors--
 
-Placeholder distractor 1
+The `break` statement
 
 ---
 
-Placeholder distractor 2
+The `label` statement
 
 ---
 
-Placeholder distractor 3
+The `return` statement
 
 #### --answer--
 
-Placeholder answer
+The `continue` statement
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+It is generally recommended to avoid using the `for...in` loop in JavaScript. What is the reason for that?
 
 #### --distractors--
 
-Placeholder distractor 1
+It only works with arrays and not with objects.
 
 ---
 
-Placeholder distractor 2
+It is the best way to iterate over large objects.
 
 ---
 
-Placeholder distractor 3
+It guarantees the order of property enumeration.
 
 #### --answer--
 
-Placeholder answer
+It could enumerate over inherited properties of an object.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What will be the output of the code below?
+
+```javascript
+for (let i = 1; i < 6; i++) {
+  if (i === 4) break;
+  console.log(i);
+}
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+```
+1,2,3,4,5,6
+```
```

Comment:
if you use a codefence it requires a language name, alternatively use a single line as below

```suggestion
`1,2,3,4,5,6`
```

Reply:
In a previous comment, I was told to add a code block since it is a terminal output. Do I change it to a single line?

Reply:
if you use a code block you need to put a language for the code block, I put both alternatives, you can use whichever you prefer